### PR TITLE
[OneShot][Testing] Add cli and oneshot pathway integration smoke tests 

### PR DESCRIPTION
**Pull request by dsikka**
# Summary
- As part of expanding overall test coverage for OneShot, introduce integration smoke tests which verify that different input combinations work for the oneshot cli command and python API
- Makes use of the updated testing framework in that every test case is defined through config files which are shared and used across the two tests (cli smoke test and API smoke test)
- Mark both tests as integration and smoke so that we run either subset through GHA
- Cadence of these tests is `commit` as the model is TinyStories and only 1 layer is targeted with 10 calibration samples

# Questions
- Seems like test type (smoke, sanity, regression) is better suited as a marker? For these test cases, I'm currently using both (marker and config field) as it may make sense to have both options available

# Testing

- You can run the following to trigger the 2 tests for each of the 4 test cases:
```bash
pytest tests/sparseml/pytorch/oneshot/ -m smoke 
```

### CODE REVIEW

**Review by Satrat — CHANGES_REQUESTED** [review]
LGTM, just a few small change requests on which args to use

File: tests/sparseml/pytorch/oneshot/test_cli.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+            "--dataset",
+            self.dataset,
+            "--model",
+            self.model,
+            "--output_dir",
+            self.output,
+            "--recipe",
+            self.recipe,
+            "--num_calibration_samples",
+            "10",
```
[the remark this answers is not in the document]
Comment (dsikka) [reply tests/sparseml/pytorch/oneshot/test_cli.py:52]: Is it toggled between True and False? Or do we always just False?

[the remark this answers is not in the document]
Comment (Satrat) [reply tests/sparseml/pytorch/oneshot/test_cli.py:52]: Should always be False for oneshot, it won't break if its true but will skew the results a bit

File: tests/sparseml/pytorch/oneshot/test_inputs.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+import pytest
+
+from parameterized import parameterized_class
+from tests.testing_utils import parse_params, requires_torch
+
+
+CONFIGS_DIRECTORY = "tests/sparseml/pytorch/oneshot/oneshot_configs"
+
+# TODO: update once this lands: https://github.com/neuralmagic/sparseml/pull/2202
```
Comment (Satrat) [inline tests/sparseml/pytorch/oneshot/test_inputs.py:26]: This just landed, lets use the example from that PR in this test

**Review by bfineran — DISMISSED** [review]

**Review by bfineran — DISMISSED** [review]

**Review by Satrat — DISMISSED** [review]

**Review by bfineran — APPROVED** [review]

**Review by Satrat — APPROVED** [review]

**Pull request merged**